components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/util/EventHookHandlerUtils.java

```diff
@@ -0,0 +1,240 @@
+import org.wso2.carbon.identity.event.IdentityEventServerException;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.identity.event.common.publisher.model.EventContext;
+import org.wso2.identity.event.common.publisher.model.EventPayload;
+import org.wso2.identity.event.common.publisher.model.SecurityEventTokenPayload;
+import org.wso2.identity.webhook.common.event.handler.constant.Constants;
+import org.wso2.identity.webhook.common.event.handler.internal.EventHookHandlerDataHolder;
+import org.wso2.identity.webhook.common.event.handler.model.EventData;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static com.wso2.identity.asgardeo.event.configuration.mgt.core.service.util.EventConfigurationMgtConstants.EVENT_CONFIG_SCHEMA_NAME_KEY;
+import static com.wso2.identity.asgardeo.event.configuration.mgt.core.service.util.EventConfigurationMgtConstants.ErrorMessages.ERROR_WHILE_CASTING_EVENT_CONFIG_AT_SERVER;
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils.CORRELATION_ID_MDC;
+
+/**
+ * This class contains the utility method implementations.
+ */
+public class EventHookHandlerUtils {
+
+    private static final Log log = LogFactory.getLog(EventHookHandlerUtils.class);
+
+    /**
+     * Retrieve event uri.
+     *
+     * @param eventKey  Event key.
+     * @return Event uri string.
+     * @throws IdentityEventServerException If an error occurs.
+     */
+    public static String getEventUri(String eventKey) throws IdentityEventServerException {
+
+        try {
+            ResourceConfig eventConfigObject = EventConfigurationMgtUtils.getEventConfig(eventKey);
+            if (eventConfigObject.getConfigs() != null &&
+                    eventConfigObject.getConfigs().containsKey(EVENT_CONFIG_SCHEMA_NAME_KEY)) {
+                return (String) eventConfigObject.getConfigs().get(EVENT_CONFIG_SCHEMA_NAME_KEY);
+            } else {
+                throw new IdentityEventServerException("Event schema not found in the resource event config " +
+                        "for the eventKey: " + eventKey);
+            }
+        } catch (EventConfigurationMgtServerException e) {
+            throw new IdentityEventServerException(e.getErrorCode(), e.getMessage());
+        } catch (ClassCastException e) {
+            throw new IdentityEventServerException(ERROR_WHILE_CASTING_EVENT_CONFIG_AT_SERVER.getCode(),
+                    ERROR_WHILE_CASTING_EVENT_CONFIG_AT_SERVER.getMessage());
+        }
+    }
+
+    /**
+     * Build the event data provider.
+     *
+     * @param event Event object.
+     * @return Event data object.
+     */
+    public static EventData buildEventDataProvider(Event event) {
+
+        Map<String, Object> properties = event.getEventProperties();
+        Map<String, Object> params = (Map)properties.get("params");
```

Comment:
Please verify get will not return a null

Reply:
Addressed with https://github.com/wso2-extensions/identity-webhook-event-handlers/pull/1/commits/6e511b91f52c4b71ccff73c989894c76b285ebf0